### ISSUE TITLE
Method to_dataframe() does not work with pynwb.misc.Units

### ISSUE THREAD

**Issue by tsgouvea** [NeurodataWithoutBorders/pynwb#1389]
## Description

Method to_dataframe() does not work with pynwb.misc.Units

Given popularity of Pandas dataframes, I believe it is desirable that this method functions.

## Steps to Reproduce
Download `https://api.dandiarchive.org/api/dandisets/000017/versions/draft/assets/3722e6b8-d47f-4feb-a9ae-9c368e41166b/` to `path_nwb`
```
import pynwb as nwb
io = nwb.NWBHDF5IO(path_nwb, 'r')
file = io.read()
u = file.units
u.to_dataframe()

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/tmp/ipykernel_324473/2196928934.py in <module>
----> 1 u.to_dataframe()

~/anaconda3/envs/myEnv/lib/python3.9/site-packages/hdmf/utils.py in func_call(*args, **kwargs)
    451 
    452             if is_method:
--> 453                 return func(args[0], **parsed['args'])
    454             else:
    455                 return func(**parsed['args'])

~/anaconda3/envs/myEnv/lib/python3.9/site-packages/hdmf/common/table.py in to_dataframe(self, **kwargs)
    564                 data[name] = [x for x in col[:]]
    565             else:
--> 566                 data[name] = col[:]
    567 
    568         return pd.DataFrame(data, index=pd.Index(name=self.id.name, data=self.id.data))

~/anaconda3/envs/myEnv/lib/python3.9/site-packages/hdmf/common/table.py in __getitem__(self, arg)
     89             ret = list()
     90             for i in indices:
---> 91                 ret.append(self.__getitem_helper(i))
     92             return ret
     93         else:

~/anaconda3/envs/myEnv/lib/python3.9/site-packages/hdmf/common/table.py in __getitem_helper(self, arg)
     82         start = 0 if arg == 0 else self.data[arg-1]
     83         end = self.data[arg]
---> 84         return self.target[start:end]
     85 
     86     def __getitem__(self, arg):

~/anaconda3/envs/myEnv/lib/python3.9/site-packages/hdmf/common/table.py in __getitem__(self, key)
    713             if isinstance(key, int) and key >= len(self.data):
    714                 raise IndexError('index {} out of bounds for data of length {}'.format(key, len(self.data)))
--> 715             return self.table[self.data[key]]
    716         else:
    717             raise ValueError("unrecognized argument: '%s'" % key)

~/anaconda3/envs/myEnv/lib/python3.9/site-packages/hdmf/common/table.py in __getitem__(self, key)
    525                     col = self.__df_cols[self.__colids[name]]
    526                     if isinstance(col.data, (Dataset, np.ndarray)) and col.data.ndim > 1:
--> 527                         data[name] = [x for x in col[arg]]
    528                     elif isinstance(col.data, np.ndarray):
    529                         data[name] = col[arg]

~/anaconda3/envs/myEnv/lib/python3.9/site-packages/hdmf/container.py in __getitem__(self, args)
    417         if isinstance(self.data, (tuple, list)) and isinstance(args, (tuple, list)):
    418             return [self.data[i] for i in args]
--> 419         return self.data[args]
    420 
    421     def append(self, arg):

h5py/_objects.pyx in h5py._objects.with_phil.wrapper()

h5py/_objects.pyx in h5py._objects.with_phil.wrapper()

~/anaconda3/envs/myEnv/lib/python3.9/site-packages/h5py/_hl/dataset.py in __getitem__(self, args, new_dtype)
    750 
    751         # Perform the dataspace selection.
--> 752         selection = sel.select(self.shape, args, dataset=self)
    753 
    754         if selection.nselect == 0:

~/anaconda3/envs/myEnv/lib/python3.9/site-packages/h5py/_hl/selections.py in select(shape, args, dataset)
     80         selector = _selector.Selector(space)
     81 
---> 82     return selector.make_selection(args)
     83 
     84 

h5py/_selector.pyx in h5py._selector.Selector.make_selection()

h5py/_selector.pyx in h5py._selector.Selector.apply_args()

TypeError: Indexing elements must be in increasing order
```

## Environment

    Python Executable: Conda
    Python Version: Python 3.9.6
    Operating System: Linux
    HDMF Version: 1.5.4
    PyNWB Version: 1.2.1

**Comment by bendichter** [comment]
Interesting, thanks for the report, @tsgouvea. This appears to be due to indexing an h5py dataset with unordered indices. e.g. doing `dataset[3,2,1]` on an `h5py.Dataset` object. I'm not exactly sure where this is arising in PyNWB but I'll have a look. Perhaps the ids of the Units table are non-ordered? They are allowed to be, but I would believe that such a case might not be tested.

**Comment by rly** [comment]
@tsgouvea Could you please try updating to the latest PyNWB? We have fixed a number of issues since the 1.2.1 release. Using the latest PyNWB 1.5.1 release, I can load the units as a pandas dataframe without error.

**Comment by tsgouvea** [comment]
I have updated to 1.5.1 and Units.to_dataframe() works as expected.
